beluga_amcl/src/amcl_node.cpp

```diff
@@ -715,14 +699,20 @@ AmclNode::CallbackReturn AmclNode::on_cleanup(const rclcpp_lifecycle::State &)
   particle_cloud_pub_.reset();
   pose_pub_.reset();
   enable_tf_broadcast_ = false;
-  last_known_estimate_.reset();
```

Comment:
@hidmic I had to remove this to make this section reachable:
https://github.com/Ekumen-OS/beluga/blob/094d7a03c34a5eda9a4717f96a34ec770c0b83e2/beluga_amcl/src/amcl_node.cpp#L858-L862

Added a test here:
https://github.com/Ekumen-OS/beluga/blob/094d7a03c34a5eda9a4717f96a34ec770c0b83e2/beluga_amcl/test/test_amcl_node.cpp#L676-L714